src/Microsoft.ML.ImageAnalytics/ImageLoaderTransform.cs

```diff
@@ -200,7 +200,11 @@ protected override Delegate MakeGetter(IRow input, int iinfo, out Action dispose
                                 // REVIEW : Log failures.
                                 dst = null;
                             }
-                        }
+							var notFound = $"File {src.ToString()} was not found";
```

Comment:
first of all: TABS, HERESY, PURGE THE HERETICS!
Now back to business.
Host.Check will throw you marked exception and stop any pipeline from execution.
Is eating exception silently bad - Yes it is.
But is, stopping whole pipeline because one of files didn't got loaded, better than it? I don't think so.

Should we send warning message to user? Maybe first few times, and then stop (it's easy to point to wrong directory) and you can end up with enormous log.

I would also probably made ImagePixelExtractor is more robust, it handle inability to process image more gracefully than just choking at format validation (just check format in If statement, and extract pixels if you can, or fallback to current logic we have for non present image) 
Also all this "Invalid character" need some proper comments in code.

Reply:
Actually, Tom and I agreed that it's better to throw than to silently pass on invalid images. We had multiple problems where incorrectly set up pipelines caused the models to misbehave, and it was very hard to debug.

---
In reply to: [223171233](https://github.com/dotnet/machinelearning/pull/1172#discussion_r223171233) [](ancestors = 223171233,223166101)

Reply:
Why we have to fall from one extreme to another?
I've work with image datasets and for every 1000 pictures you will have one which will give you a problem with loading because of format zoo and general how computational heavy image processing is.

I agree it's hard to debug, but can we at least warn user instead of just throw.

Also do we still plan to support batch prediction and how this change will affect it? Do we just stop prediction for batch if one image failed?

Also why we can't handle this case in predictor/trainer? Can we fix PixelExtractor instead and pass null to getter in case if you fail for loader?

---
In reply to: [223171285](https://github.com/dotnet/machinelearning/pull/1172#discussion_r223171285) [](ancestors = 223171285,223171233,223166101)

Reply:
I agree w/ @Ivanidzo4ka. I've run into the silent failure mode of our current image pipeline, and that extreme is terrible. 

Throwing an error is bad for the user experience. If I received an error on one image, I would simply delete it and re-run; the next step is I'll get another error, until I spend multiple hours _slowly_ finding any non-parsable files or give up.

My normal experience is a few bad images among thousands, if I knew about them, I would simply remove them. When an image fails to load/parse, I would expect it to be handled much like parse errors on other datatypes (like "abc" in a R4 column), where we warn and continue. Dirty data is normal, and images are no exception. 
 
> In reply to: [223171285](https://github.com/dotnet/machinelearning/pull/1172#discussion_r223171285) [](ancestors = 223171285,223171233,223166101)

Reply:
I feel like the issue with simply warning the user is how many other cases we then have to handle in the future to prevent an undocumented error from popping up. Specifically, a corrupted loaded image will crash in the PixelExtractor, which can certainly be fixed. But a null image (one that was not found), will pass through the PixelExtractor just fine, and won't cause a crash until the ONNX Transform which will give a .NET Core crash, which is obviously just a worse version of throwing an exception. 

If we could agree to some error statement that would prevent crashing and make sense, I am all for it, but I am not sure what it would be. My best guess would be just loading an image of all 0 values (what TLC used to do when it didn't find an image), but that also doesn't really make sense since we have to assume what the user is thinking.